components/modelregistry/modelregistry.go

```diff
@@ -45,6 +44,11 @@ var _ components.ComponentInterface = (*ModelRegistry)(nil)
 // +kubebuilder:object:generate=true
 type ModelRegistry struct {
 	components.Component `json:""`
+
+	// Namespace for model registries to be installed, configurable once, defaults to "odh-model-registries"
+	// +kubebuilder:default="odh-model-registries"
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="RegistriesNamespace is immutable"
+	RegistriesNamespace string `json:"registriesNamespace,omitempty"`
```

Comment:
why we set omitempty here?
why not `RegistriesNamespace string json:"registriesNamespace"``

then when we create https://github.com/opendatahub-io/opendatahub-operator/blob/602b5bb49bd018ddff126ac9138db0856b73efb4/tests/e2e/helper_test.go#L170 without set any value, it will get the one from default as `odh-model-registries `

Reply:
with https://github.com/dhirajsb/opendatahub-operator/commit/b2b08b582ae8a24ae660c427a6696b28e53648e3 change. 
test with this DSC to simulate our e2e test 
```
kind: DataScienceCluster
apiVersion: datasciencecluster.opendatahub.io/v1
metadata:
  labels:
    app.kubernetes.io/created-by: opendatahub-operator
    app.kubernetes.io/instance: default
    app.kubernetes.io/managed-by: kustomize
    app.kubernetes.io/name: datasciencecluster
    app.kubernetes.io/part-of: opendatahub-operator
  name: default-dsc
spec:
  components:
    modelregistry:
      managementState: Managed
```
after DSC is created,
```
spec:
  components:
    codeflare: {}
    kserve:
      serving:
        ingressGateway:
          certificate:
            type: OpenshiftDefaultIngress
        managementState: Managed
        name: knative-serving
    modelregistry:
      managementState: Managed
      registriesNamespace: odh-model-registries
    trustyai: {}
    ray: {}
    kueue: {}
    workbenches: {}
    dashboard: {}
    modelmeshserving: {}
    datasciencepipelines: {}
    trainingoperator: {}
```